src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -635,64 +635,69 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
                 ChipLogDetail(Controller, "QueryImage: application responded with: %s",
                     [[data description] cStringUsingEncoding:NSUTF8StringEncoding]);
 
-                Commands::QueryImageResponse::Type response;
-                ConvertFromQueryImageResponseParms(data, response);
-
                 auto hasUpdate = [data.status isEqual:@(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable)];
                 auto isBDXProtocolSupported = [commandParams.protocolsSupported
                     containsObject:@(MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous)];
 
-                if (hasUpdate && isBDXProtocolSupported) {
-                    auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
-                    auto nodeId = handler->GetSubjectDescriptor().subject;
-                    CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        if (err == CHIP_ERROR_BUSY) {
-                            Commands::QueryImageResponse::Type busyResponse;
-                            busyResponse.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
-                            busyResponse.delayedActionTime.SetValue(response.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
-                            handler->AddResponse(cachedCommandPath, busyResponse);
-                            handle.Release();
-                            return;
-                        }
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
-                    auto targetNodeId
-                        = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+                // If the protocol requested is not supported, return status - Protocol Not Supported
+                if (!isBDXProtocolSupported) {
```

Comment:
Kept this at the top. if the requestor sends all supported protocols then it makes sense to have it here. also if we do not care about the order of not available vs protocol not supported, programatticaly we don't need to build the delegate response from data.

Reply:
am going to write those tests now. mostly for testing the different delegate responses and simulating errors after preparefortransfer